nixos/modules/programs/sway.nix

```diff
@@ -133,8 +134,17 @@ in {
         '';
       };
     };
-    security.polkit.enable = true;
-    security.pam.services.swaylock = {};
+    security = {
+      wrappers.sway = {
+        program = "sway-setcap";
+        source = "${swayPackage.unwrapped}/bin/sway";
+        capabilities = "cap_sys_nice+ep";
```

Comment:
Due to how these wrappers within NixOS work we'll need an additional patch for sway where the ambient capability is dropped. Otherwise applications like steam won't start since they expect no capabilities at all.
I did this by adding a oneliner within `set_rr_scheduling()`:
```
prctl(PR_CAP_AMBIENT, PR_CAP_AMBIENT_CLEAR_ALL, 0, 0, 0);
```
this will also require an include of `<sys/prctl.h>`.

Reply:
What? Why? Shouldn't sway already do that itself?

Reply:
Thanks for the heads-up. I'll need to take a closer look at that (I haven't tested the capabilities part yet).

Looking at your implementation (https://github.com/swaywm/sway/pull/6994/files#diff-055d9d833b1db9e8fa249150a4d00918c36e78fb9e2e130e5663a4a6d73cbd07R13), the children/forked processes set the scheduling priority to 0 but AFAIK that shouldn't require the capability as the priority is decreased (by increasing the value). We therefore currently don't set the capability as inheritable so the forked processes shouldn't inherit any capabilities but I didn't check that either yet (maybe the NixOS abstraction for wrappers does interfere with that).

Reply:
I decided to move this into a dedicated PR after having a closer look (as it requires more testing/discussion): https://github.com/NixOS/nixpkgs/pull/207842

Reply:
> What? Why? Shouldn't sway already do that itself?

Sway expects `CAP_SYS_NICE` to be set on the binary. However this doesn't work with the way NixOS builds its binaries. Therefore NixOS employs the wrapper infrastructure, which raises the required capability to the ambient set so started processes inherit the capability. Which also means that as long as the capability remains in the ambient set, all programs started by sway will also have the capability in their ambient set. This doesn't work with jails like bubblewrap or the separation enforced by steam (which will refuse to start as long as a capability is in its ambient set). Since no other distribution requires the wrapper architecture employed by NixOS, this is a NixOS specific problem and should be fixed via a NixOS specific patch.